src/app/logged-in/admin/admin.component.ts

```diff
@@ -9,13 +10,22 @@ import { Router } from '@angular/router';
 export class AdminComponent implements OnInit {
 
   selectedTabIndex: number = 0;
-  constructor(private router: Router) { }
+  constructor(private exportService: ExportService, private dialogService: DialogService) { }
 
   ngOnInit() {
   }
 
-  changeTab(tabnum){
-    this.selectedTabIndex=tabnum;
+  changeTab(tabnum) {
+    this.selectedTabIndex = tabnum;
+  }
+
+  exportData() {
+    try {
+      this.exportService.export();
+    } catch (e) {
+      this.dialogService.confirm("Unable to download file", "Error occured generating export file");
```

Comment:
In what situation would the export fail? Should we log the errors somewhere, or make this error more detailed?

Reply:
We could probably just console log the error.